### PR TITLE
chore(maintenance): general maintencance and ci flow

### PR DESCRIPTION
**Pull request by peterdeme**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            core.exportVariable('VERSION', version)
+
+      - uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+
+      - name: Publish to PyPi
+        env:
+          TWINE_USERNAME: "__token__"
+          TWINE_PASSWORD: "${{ secrets.PYPI_TOKEN }}"
```
Comment (peterdeme) [inline .github/workflows/release.yml:37]: Can you add it @ferhatelmas? 🙏

**Comment by peterdeme** [comment]
Can you remove the yellow ones @ferhatelmas? 🙏 

image

File: .github/workflows/release.yml (hunk 1 above)
Comment (ferhatelmas) [reply .github/workflows/release.yml:37]: had added before

File: .github/workflows/reviewdog.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: reviewdog/action-setup@v1
+        with:
+          reviewdog_version: latest
+
+      - uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
```
Comment (ferhatelmas) [inline .github/workflows/reviewdog.yml:22]: using oldest runtime might be better

Comment (peterdeme) [reply .github/workflows/reviewdog.yml:22]: Why? 🤔

Comment (ferhatelmas) [reply .github/workflows/reviewdog.yml:22]: so as to check if we are fine with old way of doing things

Comment (ferhatelmas) [reply .github/workflows/reviewdog.yml:22]: similar to your change in php (7.3 instead of 7.4)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -5,33 +5,40 @@ on:
...
+  cancel-in-progress: true
+
 jobs:
   build:
+    name: 🧪 Test & lint
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python: [3.6, 3.7, 3.8, 3.9]
+        python: ["3.7", "3.8", "3.9", "3.10"]
```
Comment (ferhatelmas) [inline .github/workflows/ci.yml:18]: this needs update in setup py too

Comment (ferhatelmas) [reply .github/workflows/ci.yml:18]: I see it's already done

**Review by ferhatelmas — APPROVED** [review]

File: .github/workflows/reviewdog.yml (hunk 2 above)
Comment (ferhatelmas) [reply .github/workflows/reviewdog.yml:22]: it's tooling, we can adjust later.

**Pull request merged**